common/statsd/statsd.cc

```diff
@@ -49,6 +49,8 @@ class StatsdClientWrapper {
     }
 
     void addMetric(string_view name, size_t value, string_view type, vector<pair<const char *, const char *>> tags) {
+        ENFORCE(link != nullptr);
```

Comment:
you could also make `link` const so that it can only be set by the constructor and then you'd know that either the constructor raised, or it was set to a non-null value, and then not have to ENFORCE at all the usages.

Reply:
the `statsd` client methods called on `link` take it as an argument and make no guarantee that they won't modify it; won't typecheck